### ISSUE TITLE
Process hangs on publish if the producer is deleted

### ISSUE THREAD

**Issue by rpoply1** [KxSystems/kafka#57]
Trying to publish a message on a topic whose associated producer is deleted hangs the process 

```
producer: .kfk.Producer enlist[`metadata.broker.list]!enlist `localhost:9011;
random: .kfk.Topic[producer;`test;()!()];
.kfk.ClientDel producer;
.kfk.Pub[random; -1i; -8!"hello world" ;""]
```

**Comment by cmccarthy1** [comment]
This is reasonably simple to fix but would require a change to the variable input to the `.kfk.Pub` function.

Essentially the `.kfk.Pub` function would also need to take in the client index. Such that 

```
.kfk.Pub[producer;random;-1i;-8"hello world";""]
```
In order to check that both the client you are publishing to and the topic are valid. This issue is arising because the topic still maintains a reference to the producer which can't be updated dynamically presently.

A discussion on how to best make this change without breaking historical behaviour will need to take place as the addition of an additional parameter to `.kfk.Pub` would break systems using legacy code. Will revert here once the change has been discussed.

**Comment by cmccarthy1** [comment]
The fix currently being worked on for #65 should also rectify this issue by removing all topics associated with a client on deletion
